src/sidebar/components/annotation-action-bar.js

```diff
@@ -10,9 +10,22 @@ import { withServices } from '../util/service-context';
 import AnnotationShareControl from './annotation-share-control';
 import Button from './button';
 
+/** @typedef {import("../../types/api").Annotation} Annotation */
+
+/**
+ * @typedef AnnotationActionBarProps
+ * @prop {Annotation} annotation - The annotation in question
+ * @prop {(Event) => any} onReply - Callbacks for when action buttons are clicked/tapped
+ * @prop {Object} annotationsService
+ * @prop {Object} settings
+ * @prop {Object} toastMessenger
+ */
+
 /**
  * A collection of `Button`s in the footer area of an annotation that take
  * actions on the annotation.
+ *
+ * @param {annotation} props
```

Comment:
The type is wrong here. This should be:

```suggestion
 * @param {AnnotationActionBarProps} props
```

I don't know why this doesn't result in an error, but the same thing appears to happen if you use any property name from the destructuring assignment (eg. `annotation`, `onReply` ...).

Making this change flags up an issue with the `flagged` prop missing from the `Annotation` type.

Reply:
thats so odd it didnt fail :\